From: =?utf-8?b?Ik1hcmNvIFRyZXZpc2FuIChUcmV2acOxbyki?= <mail@3v1n0.net>
Date: Wed, 21 Jan 2026 20:19:36 +0100
Subject: gio/gcontenttype-fdo: Do not overflow if header is longer than
 MAXINT

In case the header size is longer than MAXINT we may read and write to
invalid locations

Spotted by treeplus.
Thanks to the Sovereign Tech Resilience programme from the Sovereign
Tech Agency.

Bug: https://gitlab.gnome.org/GNOME/glib/-/issues/3871
Bug-CVE: CVE-2026-1485
Bug-Debian: https://bugs.debian.org/1126550
Applied-upstream: 2.87.3, commit:aacda5b07141b944408c79e83bcbed3b2e1e6e45
---
 gio/gcontenttype-fdo.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gio/gcontenttype-fdo.c b/gio/gcontenttype-fdo.c
index 230cea1..1132397 100644
--- a/gio/gcontenttype-fdo.c
+++ b/gio/gcontenttype-fdo.c
@@ -817,7 +817,7 @@ tree_match_free (TreeMatch *match)
 static TreeMatch *
 parse_header (gchar *line)
 {
-  gint len;
+  size_t len;
   gchar *s;
   TreeMatch *match;
 
